src/components/Transaction/Form.tsx

```diff
@@ -0,0 +1,199 @@
+import PropTypes from 'prop-types'
+import { useState, ReactNode, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { Message } from '@glif/filecoin-message'
+import { FilecoinNumber, BigNumber } from '@glif/filecoin-number'
+import {
+  FILECOIN_NUMBER_PROPTYPE,
+  MESSAGE_PROPTYPE,
+  TxState,
+  TX_STATE_PROPTYPE
+} from '../../customPropTypes'
+import { MessagePending } from '../../generated/graphql'
+import { TransactionButtons } from './Buttons'
+import { TransactionFee } from './Fee'
+import { TransactionHeader } from './Header'
+import { TransactionTotal } from './Total'
+import { Dialog, ShadowBox } from '../Layout'
+import { useSubmittedMessages } from '../HistoryTables/PendingMsgContext'
+import { getMaxGasFee } from '../../utils'
+import { useWallet, useWalletProvider } from '../../services'
+import { logger } from '../../logger'
+
+export const TransactionForm = ({
+  children,
+  title,
+  description,
+  message,
+  total,
+  txState,
+  setTxState,
+  maxFee,
+  txFee,
+  setTxFee,
+  getParams,
+  onComplete
+}: TransactionFormProps) => {
+  const router = useRouter()
+  const wallet = useWallet()
+  const { pushPendingMessage } = useSubmittedMessages()
+  const { loginOption, walletError, getProvider } = useWalletProvider()
+
+  // Transaction states
+  const [txError, setTxError] = useState<Error | null>(null)
+  const [gasParamsError, setGasParamsError] = useState<Error | null>(null)
+  const [messageWithGas, setMessageWithGas] = useState<Message | null>(null)
+
+  // Max transaction fee used for getting gas params. Will be
+  // null until the user manually changes the transaction fee.
+  const [inputFee, setInputFee] = useState<FilecoinNumber | null>(null)
+
+  // Calculate transaction fee (fee cap times limit)
+  useEffect(() => {
+    if (!messageWithGas) return setTxFee(null)
+    const feeCapStr = messageWithGas.gasFeeCap.toFixed(0, BigNumber.ROUND_CEIL)
+    const feeCap = new FilecoinNumber(feeCapStr, 'attofil')
+    const limit = messageWithGas.gasLimit
+    setTxFee(getMaxGasFee(feeCap, limit))
+  }, [setTxFee, messageWithGas])
```

Comment:
And FYI, the reason that `Send.tsx` needs to access the wallet is because it determines the Total value, which is specific per implementation. In the wallet, total value for Withdraw does _not_ include the tx fee